packages/react-native/tests/storage/FileBreadcrumbsStorage.spec.ts

```diff
@@ -0,0 +1,415 @@
+                attributes: {},
+            },
+        ];
+
+        const expectedMain: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'warning',
+                message: 'c\n3',
+                timestamp: expect.any(Number),
+                type: 'navigation',
+                attributes: {},
+            },
+        ];
+
+        const expectedFallback: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'debug',
+                message: 'b\n2',
+                timestamp: expect.any(Number),
+                type: 'http',
+            },
+        ];
+
+        const storage = new FileBreadcrumbsStorage(session, fs, {
+            maximumBreadcrumbs: 2,
+        });
+
+        for (const breadcrumb of breadcrumbs) {
+            storage.add(breadcrumb);
+            await nextTick();
+        }
+
+        // FileBreadcrumbsStorage is asynchronous in nature
+        await nextTick();
+
+        const [fallbackAttachment, mainAttachment] = storage.getAttachments();
+
+        const mainStream = mainAttachment.get();
+        const fallbackStream = fallbackAttachment.get();
+        assert(mainStream);
+        assert(fallbackStream);
+
+        const actualMain = await loadBreadcrumbs(fs, mainStream);
+        const actualFallback = await loadBreadcrumbs(fs, fallbackStream);
+        expect(actualMain).toEqual(expectedMain);
+        expect(actualFallback).toEqual(expectedFallback);
+    });
+
+    it('should return breadcrumbs up to the json size', async () => {
+        const fs = mockStreamFileSystem();
+        const session = new SessionFiles(fs, '.', 'sessionId');
+
+        const breadcrumbs: RawBreadcrumb[] = [
+            {
+                level: BreadcrumbLogLevel.Debug,
+                message: 'a',
+                type: BreadcrumbType.Http,
+            },
+            {
+                level: BreadcrumbLogLevel.Debug,
+                message: 'b',
+                type: BreadcrumbType.Http,
+            },
+            {
+                level: BreadcrumbLogLevel.Debug,
+                message: 'c',
+                type: BreadcrumbType.Http,
+            },
+        ];
+
+        const expectedMain: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'debug',
+                message: 'c',
+                timestamp: expect.any(Number),
+                type: 'http',
+            },
+        ];
+
+        const expectedFallback: Breadcrumb[] = [
+            {
+                id: expect.any(Number),
+                level: 'debug',
+                message: 'b',
+                timestamp: expect.any(Number),
+                type: 'http',
+            },
+        ];
+
+        const storage = new FileBreadcrumbsStorage(session, fs, {
+            maximumBreadcrumbs: 100,
+            maximumTotalBreadcrumbsSize: (JSON.stringify(expectedMain[0]).length + 10) * 2,
```

Comment:
wouldn't it be better if you can sum lenght of expectedMain[0] + expectedMain[1] length instead of doing `*2` If you modify the test, it might be broken

Reply:
good catch, changed.